### PR TITLE
#84 Feature Dependency and Impact Analysis: Impact Analysis & Data Retrieval API

### PR DESCRIPTION
**Pull request by DmitriyChernyakSetronica**
**Objective**
Develop the backend logic and API endpoints to retrieve dependency information and analyze the impact of feature changes. 🔍

**Requirements**
* **Dependency Retrieval:** Implement logic to fetch all dependencies (features this one relies on) and dependents (features that rely on this one).
* **Impact Analysis Logic:** Create a service that can identify all features potentially affected by a change to a given feature, including both direct and indirect dependents.
* **Filtering:** Support filtering the impact analysis results by parameters such as product, release, or status.

**API Endpoints** are annotated with @transactional to ensure proper session management for lazy loading:                 
* `GET /features/{featureCode}/dependencies`: List all features that `{featureCode}` depends on.
* `GET /features/{featureCode}/dependents`: List all features that depend on `{featureCode}`.
* `GET /features/{featureCode}/impact`: Provide a comprehensive list of all affected features, with support for filtering via query parameters.
* Response format returns FeatureDependencyDto objects with JSON structure:
`{
  "id": 1,
  "feature": { "code": "FEATURE-A", "title": "...", ... },
  "dependsOnFeature": { "code": "FEATURE-B", "title": "...", ... },
  "dependencyType": "HARD|SOFT|OPTIONAL",
  "notes": "Optional description",
  "createdAt": "2024-01-01T00:00:00Z"
}`

**Test Coverage**
* Tests for correct identification of all direct and indirect dependents.
* Tests for API endpoints, including the filtering logic.
* Tests for edge cases, such as circular dependencies or features with no dependents.

**Acceptance Criteria**
* The API correctly returns all dependencies and dependents for a given feature.
* The impact analysis endpoint accurately identifies the full scope of affected features.
* Filtering works as specified.

FAIL_TO_PASS: FeatureDependencyEndpointsIntegrationTests.java

### CODE REVIEW

File: src/test/java/com/sivalabs/ft/features/api/controllers/FeatureDependencyEndpointsIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+                .bodyJson()
+                .extractingPath("$.size()")
+                .asNumber()
+                .isEqualTo(1);
+
+        // Verify the returned dependency
+        assertThat(result)
+                .bodyJson()
+                .extractingPath("$[0]")
+                .convertTo(FeatureDependencyDto.class)
```
Comment (volkov) [inline src/test/java/com/sivalabs/ft/features/api/controllers/FeatureDependencyEndpointsIntegrationTests.java:104]: May be we should not use new dto in tests, though here it's pretty obvious that this Dto should work.

We for example could change methods to return Feature? Then we can use existing feature dto.

File: src/test/java/com/sivalabs/ft/features/api/controllers/FeatureDependencyEndpointsIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+        // Then - Verify circular dependency exists
+        assertThat(impactBefore)
+                .hasStatusOk()
+                .bodyJson()
+                .extractingPath("$.size()")
+                .asNumber()
+                .isEqualTo(2); // Should include IDEA-2 and GO-3
+
+        // When - Delete GO-3 feature through API
+        var deleteResult = mvc.delete().uri("/api/features/{code}", "GO-3").exchange();
```
Comment (volkov) [inline src/test/java/com/sivalabs/ft/features/api/controllers/FeatureDependencyEndpointsIntegrationTests.java:447]: Do we really need to test deletions?
